libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/guidance/GuidanceViewUrlMatcher.kt

```diff
@@ -0,0 +1,32 @@
+@file:JvmName("GuidanceViewUrlMatcher")
+
+package com.mapbox.navigation.ui.maps.guidance
+
+import com.mapbox.api.directions.v5.models.BannerComponents
+
+internal fun String.guidanceViewUrlMatcher(
+    type: String,
+    accessToken: String,
+    key: String = "access_token=",
+): String {
+    when (type) {
+        BannerComponents.JCT -> {
+            return if (this.endsWith('&')) {
+                this.plus(key.plus(accessToken))
+            } else if (this.endsWith('?')) {
+                this.replace('?', '&').plus(key.plus(accessToken))
+            } else {
+                this.plus("&".plus(key).plus(accessToken))
```

Comment:
This is incorrect. We are not verifying whether there are any existing query parameters, and rarely would a URL end with `?` or `&`.